src/tests/ftest/daos_test/daos_core_test_dfs.py

```diff
@@ -36,16 +35,36 @@ def __init__(self, *args, **kwargs):
         """Initialize the DaosCoreBase object."""
         super(DaosCoreTestDfs, self).__init__(*args, **kwargs)
         self.hostfile_clients_slots = None
+        self.daos_test = os.path.join(self.bin, 'dfs_test')
```

Comment:
This assignment must be made in `setUp()` (after calling `super(<class>, self).setUp()`) or in a `test_*()` method as `self.bin` is not assigned until https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/apricot/apricot/test.py#L318.  As a result we hit this error:
```
[2021-01-27T21:19:46.228Z] Running: avocado run --ignore-missing-references on --html-job-result on --tap-job-result off --filter-by-tags=pr,hw,large --mux-yaml /tmp/tmpjs3m8r/daos_test-daos_core_test_dfs.yaml -- ./daos_test/daos_core_test_dfs.py
[2021-01-27T21:19:46.486Z] JOB ID     : 9346e8e418eb973cbf87f8816f29113fa6250ac6
[2021-01-27T21:19:46.486Z] JOB LOG    : /var/tmp/ftest/avocado/job-results/job-2021-01-27T21.19-9346e8e/job.log
[2021-01-27T21:19:47.072Z] Traceback (most recent call last):
[2021-01-27T21:19:47.072Z] 
[2021-01-27T21:19:47.072Z]   File "/usr/lib/python2.7/site-packages/avocado/core/job.py", line 453, in run_tests
[2021-01-27T21:19:47.072Z]     execution_order)
[2021-01-27T21:19:47.072Z] 
[2021-01-27T21:19:47.072Z]   File "/usr/lib/python2.7/site-packages/avocado/core/runner.py", line 599, in run_suite
[2021-01-27T21:19:47.072Z]     deadline):
[2021-01-27T21:19:47.072Z] 
[2021-01-27T21:19:47.072Z]   File "/usr/lib/python2.7/site-packages/avocado/core/runner.py", line 393, in run_test
[2021-01-27T21:19:47.072Z]     test_status.wait_for_early_status(proc, 60)
[2021-01-27T21:19:47.072Z] 
[2021-01-27T21:19:47.072Z] 
[2021-01-27T21:19:47.072Z] Avocado crashed: TestError: Process died before it pushed early test_status.
```
Due to:
```
2021-01-27 21:19:46,801 output           L0655 DEBUG| Traceback (most recent call last):
2021-01-27 21:19:46,802 output           L0655 DEBUG|   File "/usr/lib64/python2.7/multiprocessing/process.py", line 258, in _bootstrap
2021-01-27 21:19:46,802 output           L0655 DEBUG|     self.run()
2021-01-27 21:19:46,802 output           L0655 DEBUG|   File "/usr/lib64/python2.7/multiprocessing/process.py", line 114, in run
2021-01-27 21:19:46,802 output           L0655 DEBUG|     self._target(*self._args, **self._kwargs)
2021-01-27 21:19:46,802 output           L0655 DEBUG|   File "/usr/lib/python2.7/site-packages/avocado/core/runner.py", line 326, in _run_test
2021-01-27 21:19:46,802 output           L0655 DEBUG|     instance = loader.load_test(test_factory)
2021-01-27 21:19:46,802 output           L0655 DEBUG|   File "/usr/lib/python2.7/site-packages/avocado/core/loader.py", line 338, in load_test
2021-01-27 21:19:46,813 output           L0655 DEBUG|     test_instance = test_class(**test_parameters)
2021-01-27 21:19:46,813 output           L0655 DEBUG|   File "/usr/lib/daos/TESTING/ftest/daos_test/daos_core_test_dfs.py", line 38, in __init__
2021-01-27 21:19:46,814 output           L0655 DEBUG|     self.daos_test = os.path.join(self.bin, 'dfs_test')
2021-01-27 21:19:46,814 output           L0655 DEBUG|   File "/usr/lib64/python2.7/posixpath.py", line 77, in join
2021-01-27 21:19:46,814 output           L0655 DEBUG|     elif path == '' or path.endswith('/'):
2021-01-27 21:19:46,814 output           L0655 DEBUG| AttributeError: 'NoneType' object has no attribute 'endswith'
```

Reply:
thanks